contracts/test/LONStaking.t.sol

```diff
@@ -248,17 +249,51 @@ contract LONStakingTest is Test {
         lonStaking.stake(DEFAULT_STAKE_AMOUNT);
     }
 
-    function testStake() public {
-        BalanceSnapshot.Snapshot memory userLon = BalanceSnapshot.take(user, address(lon));
+    function _stakeAndValidate(address staker, uint256 stakeAmount) internal {
+        BalanceSnapshot.Snapshot memory stakerLon = BalanceSnapshot.take(staker, address(lon));
         BalanceSnapshot.Snapshot memory lonStakingLon = BalanceSnapshot.take(address(lonStaking), address(lon));
-        BalanceSnapshot.Snapshot memory userXLON = BalanceSnapshot.take(user, address(lonStaking));
-        uint256 stakeAmount = DEFAULT_STAKE_AMOUNT;
+        BalanceSnapshot.Snapshot memory stakerXLON = BalanceSnapshot.take(staker, address(lonStaking));
         uint256 expectedStakeAmount = _getExpectedXLON(stakeAmount);
-        vm.prank(user);
+        vm.startPrank(staker);
+        if (lon.allowance(staker, address(lonStaking)) == 0) {
+            lon.approve(address(lonStaking), type(uint256).max);
+        }
         lonStaking.stake(stakeAmount);
-        userLon.assertChange(-int256(stakeAmount));
+        stakerLon.assertChange(-int256(stakeAmount));
         lonStakingLon.assertChange(int256(stakeAmount));
-        userXLON.assertChange(int256(expectedStakeAmount));
+        stakerXLON.assertChange(int256(expectedStakeAmount));
+        vm.stopPrank();
+    }
+
+    function testStake() public {
+        _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT);
+    }
+
+    function testFuzz_Stake(uint256 stakeAmount) public {
+        vm.assume(stakeAmount > 0);
+        vm.assume(stakeAmount <= lon.cap());
+        vm.assume(stakeAmount.add(lon.totalSupply()) <= lon.cap());
+
+        lon.mint(user, stakeAmount);
+        _stakeAndValidate(user, stakeAmount);
+    }
+
+    function testFuzz_StakeMultiple(uint80[16] memory stakeAmounts) public {
```

Comment:
The number `16` is not justified. It can be increased and depends on how much time are we willing to spend on generating valid inputs.

Reply:
So foundry will base on the argument type to generate valid inputs, including array with multiple dimension, right?

Reply:
Yes!

Reply:
Super powerful! 🤩